src/imageclipper.cpp

```diff
@@ -43,835 +44,745 @@
-                exit(1);
-            }
-
-            if ( (param->filelist.begin() + arg->frame) > param->filelist.end() )
-            {
-                usage( arg );
-                cerr << "Specified frame " << arg->frame << " is past the end of the list of "
-                     << param->filelist.size() << " files " << endl;
-                exit(1);
-            }
-
-            param->fileiter = param->filelist.begin() + arg->frame;
-        }
-        else
-        {
-            if( !fs::exists( arg->reference ) )
-            {
-                usage( arg );
-                cerr << "The image file " << fs::realpath( arg->reference ) << " does not exist." << endl << endl;
-                exit(1);
-            }
-            param->filelist = fs::filelist( fs::dirname( arg->reference ), param->imtypes, "file" );
-
-            // step up till specified file
-            for( param->fileiter = param->filelist.begin(); param->fileiter != param->filelist.end(); param->fileiter++ )
-            {
-                if( fs::realpath( *param->fileiter ) == fs::realpath( arg->reference ) ) break;
-            }
-        }
-
-        cerr << "Done!" << endl;
-        cerr << arg->frame << " Now showing " << fs::realpath( *param->fileiter ) << endl;
-        param->img = cvLoadImage( fs::realpath( *param->fileiter ).c_str() );
-    }
-    else if( is_video )
-    {
-        if ( !fs::exists( arg->reference ) )
-        {
-            usage( arg );
-            cerr << "The file " << fs::realpath( arg->reference ) << " does not exist or is not readable." << endl << endl;
-            exit(1);
-        }
-        cerr << "Now reading a video..... ";
-        param->cap = cvCaptureFromFile( fs::realpath( arg->reference ).c_str() );
-        cvSetCaptureProperty( param->cap, CV_CAP_PROP_POS_FRAMES, arg->frame - 1 );
-        param->img = cvQueryFrame( param->cap );
-        if( param->img == NULL )
-        {
-            usage( arg );
-            cerr << "The file " << fs::realpath( arg->reference ) << " was assumed as a video, but not loadable." << endl << endl;
-            exit(1);
-        }
-        cerr << "Done!" << endl;
-        cerr << cvGetCaptureProperty( param->cap, CV_CAP_PROP_FRAME_COUNT ) << " frames totally." << endl;
-        cerr << " Now showing " << fs::realpath( arg->reference ) << " " << arg->frame << endl;
-    }
-    else
-    {
-        usage( arg );
-        cerr << "The directory " << fs::realpath( arg->reference ) << " does not exist." << endl << endl;
-        exit(1);
-    }
+	//system("mode con lines=50 cols=80");
+
+	// initialization
+	CvCallbackParam init_param = {
+		"<S> Save <F> Forward <SPACE> s and f <B> Backward <ESC> Exit",			// main window's name
+		"Cropped",																// cropped window's name
+		NULL,																	// source image pointer
+		cvRect(0, 0, 0, 0),														// selected rect
+		cvRect(0, 0, 0, 0),														// watershed
+		0,																		// rotate

```

Comment:
You seem to have a very wide screen, all these comments are way off screen on mine. Could you try to keep the   line length below 120 chars?